### ISSUE TITLE
Utilize record search in client factory methods

### ISSUE THREAD

**Issue by Comeani** [Better-HPC/keystone-python-client#64]
Right now the various factory methods use a filter parameter to truncate data gathered from the server down to the required bits. Once https://github.com/Better-HPC/keystone-api/issues/478 in place, we can consider how we might utilize that support in the relevant client operations.

**Comment by djperrefort** [comment]
@Comeani is this closed by #65

**Comment by Comeani** [comment]
Yes, I'll go take a look and see where using these might be applicable to the wrappers.
